### PR TITLE
Integration - property.py - Initial integration and unit tests for property.py

### PR DESCRIPTION
**Pull request by emmanesbit**
Initial tests to cover most of the functionality prior to any refactoring.

### CODE REVIEW

**Comment by emmanesbit** [comment]
@cflynn3 @andy-beer - This PR adds some extra tests for property.py, and a few very minor bug fixes. I'll add some additional tests after refactoring property.py into separate files

File: resqpy/property.py [hunk 1]
```diff
@@ -2724,8 +2725,8 @@ def add_cached_array_to_imported_list(self,
             if not discrete and np.all(np.isnan(zorro)):
                 min_value = max_value = None
             else:
-                min_value = np.nanmin(zorro)
-                max_value = np.nanmax(zorro)
+                min_value = int(np.nanmin(zorro))
```
Comment (andy-beer) [inline resqpy/property.py:2728]: The logic is not quite right here.  The int() must only be applied when discrete is True.

**Review by andy-beer — COMMENTED** [review]
The tests look excellent.  I think there is still something wrong with the min_value, max_value code.

**Comment by emmanesbit** [comment]
@andy-beer good spot on the min/max logic - this should now be resolved (and perhaps suggests another test is needed in the next PR to check min/max for non-integer continuous properties!)

File: resqpy/property.py (hunk 1 above)
Comment (andy-beer) [reply resqpy/property.py:2728]: I think that will do it, so I'll mark this as resolved.  Could I suggest a test with a discrete array, with a null value defined (typically -1) and all the values in the array set to the null value.  Ideally the min and max values would end up as None in that scenario.  If it comes out different, we might need to strip out the int(np.nanmin()) and use a simple np.min() on the masked zorro array.

**Review by andy-beer — APPROVED** [review]

**Review by cflynn3 — APPROVED** [review]

**Pull request merged**